library/runtime/citizen.lua

```diff
@@ -2,34 +2,146 @@
+
+---Retrieves an initialized float value from a pointer. Same as `Citizen.PointerValueInt`, but this assumes the pointer is initialized.
+---@param pointer integer; The pointer to the address in the memory.
+---@return userdata; The data retrieved by the pointer.
+function Citizen.PointerValueIntInitialized(pointer) end
+
+---Retrieves an initialized integer value from a pointer. Same as `Citizen.PointerValueFloat`, but this assumes the pointer is initialized.
+---@param pointer integer; The pointer to the address in the memory.
+---@return userdata; The data retrieved by the pointer.
+function Citizen.PointerValueFloatInitialized(pointer) end
+
+---Retrieves an integer value from a pointer.
+---@return userdata; The data retrieved by the pointer.
+function Citizen.PointerValueInt() end
+
+---Retrieves a float value from a pointer.
+---@return userdata; The data retrieved by the pointer.
+function Citizen.PointerValueFloat() end
+
+---Retrieves a vector value from a pointer.
+---@return userdata; The data retrieved by the pointer.
+function Citizen.PointerValueVector() end
+
+---Specifies that a result should be returned regardless of success.
+---@return any; The result.
+function Citizen.ReturnResultAnyway() end
+
+---Converts the result of a native invocation to an integer.
+---@return integer; The result as an integer.
+function Citizen.ResultAsInteger() end
+
+---Converts the result of a native invocation to a long integer.
+---@return integer; The result as a long integer.
+function Citizen.ResultAsLong() end
+
+---Converts the result of a native invocation to a float.
+---@return number; The result as a float.
+function Citizen.ResultAsFloat() end
+
+---Converts the result of a native invocation to a string.
+---@return string; The result as a string.
+function Citizen.ResultAsString() end
+
+---Converts the result of a native invocation to a vector.
+---@return vector; The result as a vector.
+function Citizen.ResultAsVector() end
+
+---@see Citizen.ResultAsObject2 Because this is unused in all native loaders.
+function Citizen.ResultAsObject() end
+
+---Converts the result of a native invocation to an object.
+---@return table; The result as an object.
+---@param msgpack_unpack function
+function Citizen.ResultAsObject2(msgpack_unpack) end
+
+---Gets a function reference for a given function. This can be used to pass functions across the network or to native invocations where a function reference is required.
+---If the input is a table with a `__cfx_functionReference` key, it wraps the table in a function call.
+---@param fun function The function or table to get a reference for.
+---@return string|nil; A string reference to the function if successful, nil otherwise.
+function Citizen.GetFunctionReference(fun) end
+
+---Converts a reference into a canonical form.
+---@param reference integer The reference to canonicalize.
+---@return string; The canonical form of the reference.
+function Citizen.CanonicalizeRef(reference) end
```

Comment:
I don't think implementation details like this are important to users; unless you have a usecase?